consul/catalog_endpoint.go

```diff
@@ -36,6 +37,17 @@ func (c *Catalog) Register(args *structs.RegisterRequest, reply *struct{}) error
 			return fmt.Errorf("Must provide service name with ID")
 		}
 
+		// Verify that the service name is valid per RFC-1123 and
+		// RFC-2782.
+		//
+		// FIXME(sean@): After 0.7 has been branched, change from a
+		// warning to an error for the 0.8 release.
+		re := regexp.MustCompile(`^(?i:[a-z0-9]|[a-z0-9][a-z0-9\-]{0,61}[a-z0-9])$`)
```

Comment:
I didn't find a place to stash the compiled regexp object.  If there is a place, lmk.  Because this object is created at service registration time, I wasn't worried about the performance impact, but it was a nit I didn't resolve.